cli/flox-rust-sdk/src/models/environment/generations.rs

```diff
@@ -673,6 +928,361 @@ mod tests {
     const GEN_ID_1: GenerationId = GenerationId(1);
     const GEN_ID_2: GenerationId = GenerationId(2);
 
+    const AUTHOR: &str = "author";
+    const HOSTNAME: &str = "host";
+
+    fn default_add_generation_options() -> AddGenerationOptions {
+        AddGenerationOptions {
+            author: AUTHOR.into(),
+            hostname: HOSTNAME.into(),
+            timestamp: Utc::now(),
+            kind: HistoryKind::Other {
+                summary: "mock".into(),
+            },
+        }
+    }
+
+    fn default_switch_generation_options(next_generation: GenerationId) -> SwitchGenerationOptions {
+        SwitchGenerationOptions {
+            author: AUTHOR.into(),
+            hostname: HOSTNAME.into(),
+            timestamp: Utc::now(),
+            next_generation,
+        }
+    }
+
+    mod metadata {
```

Comment:
I've been thinking a few times that `proptest` could be really useful here.
I couldn't really tell if that's true and wouldn't duplicate logic, or how to correctly add proptest in the mix, so for now its explicit test cases.